src/Service/DistanceMatrix/Request/DistanceMatrixRequest.php

```diff
@@ -504,7 +505,7 @@ public function setLanguage($language = null)
     public function buildQuery()
     {
         $locationBuilder = function (LocationInterface $location) {
-            return $location->buildQuery();
+            return $location->buildQuery() . ($location instanceof EncodedPolylineLocation ? ':' : '');
```

Comment:
@dergus Ok, I will give a try to your branch and look why it's not working (IMO, it is related to the API key used by the tests (which has been revoked).